### DISCUSSION TITLE
Option object for WithMiddleware.

### DISCUSSION THREAD

**Discussion by joelmandell** [umbraco/Umbraco-CMS#14605] [Features and ideas]
Hi ;)

I have an idea for adding an option/config option for WithMiddleware to add support for cors policies. I can add an PR if that is something you think could be useful. Instead of doing an WithCustomMiddleware and calling all builders manually, one could have a config/option object with for example an CorsPolicy array, that adds cors policies in correct order (after UseRouting and before UseAuthentication and so on).

What do you guys think about that?

**Comment by nul800sebastiaan** [comment]
This PR is upcoming for 12.1, which obviates the need for calling all builders manually: https://github.com/umbraco/Umbraco-CMS/pull/14503

**Comment by bergmania** [comment]
Because https://github.com/umbraco/Umbraco-CMS/pull/14503 is now released I will close this discussion
